tests/test_geoprocessing.py

```diff
@@ -4186,3 +4186,40 @@ def test_convolve_2d_nodata(self):
         expected_output[0, 0] = 1
         expected_output[-1, -1] = 1
         numpy.testing.assert_allclose(target_array, expected_output)
+
+    def test_convolve_2d_gaussian(self):
+        """PGP.geoprocessing: test convolve 2d with large gaussian kernel."""
+        # choosing twice the max memory block
+        n_pixels = 256*2
+        # this is a fun seed random seed
+        numpy.random.seed(n_pixels)
+        signal_array = numpy.random.random((n_pixels, n_pixels))
+
+        workspace_dir = 'local'
+        if not os.path.exists(workspace_dir):
+            os.makedirs(workspace_dir)
+
+        base_nodata = -1
+        signal_path = os.path.join(workspace_dir, 'signal.tif')
+        _array_to_raster(signal_array, base_nodata, signal_path)
+
+        kernel_seed = numpy.zeros((n_pixels, n_pixels))
+        kernel_seed[n_pixels//2, n_pixels//2] = 1
+        kernel_array = scipy.ndimage.gaussian_filter(kernel_seed, 1.0)
+        kernel_path = os.path.join(workspace_dir, 'kernel.tif')
+        _array_to_raster(kernel_array, base_nodata, kernel_path)
+
+        target_path = os.path.join(workspace_dir, 'target.tif')
+        pygeoprocessing.convolve_2d(
+            (signal_path, 1), (kernel_path, 1), target_path,
+            ignore_nodata_and_edges=False, mask_nodata=True)
+        target_array = pygeoprocessing.raster_to_numpy_array(target_path)
+
+        # gaussian filter with constant is the same as bleeding off the edges
+        expected_output = scipy.ndimage.gaussian_filter(
+            signal_array, 1.0, mode='constant')
+
+        _array_to_raster(
+            expected_output, None, os.path.join(workspace_dir, 'extected.tif'))

```

Comment:
Not a big deal, but was this supposed to be `expected.tif`?

Reply:
Also, is there any reason to save this array to a raster?

Reply:
Nope! Same as above, left over from testing/debugging. Removed in 3a5f174093f133daeb8f60de269d8220a2341971